mito-ai/src/Extensions/AppBuilder/auth-popup-deploy.tsx

```diff
@@ -0,0 +1,102 @@
+// auth-popup.tsx - Simple authentication popup
+import React from 'react';
+import { Authenticator } from '@aws-amplify/ui-react';
+import '@aws-amplify/ui-react/styles.css';
+import '../../../style/ConnectionForm.css';
+import '../../../style/button.css';
+import '../../../style/AuthPopup.css';
+import TextButton from '../../components/TextButton';
+
+interface AuthPopupProps {
+  isOpen: boolean;
+  onClose: () => void;
+  onSuccess: (user: any) => void;
+}
+
+// Using Mito's modal styling classes
+
+export const AuthPopup: React.FC<AuthPopupProps> = ({
+  isOpen,
+  onClose,
+  onSuccess
+}) => {
+  // Track if we've already called onSuccess to prevent infinite loops
+  const hasCalledOnSuccess = React.useRef(false);
+
+  if (!isOpen) return null;
+
+  return (
+    <div className="modal-overlay">
+      <div className="modal-content">
+        <div className="modal-header">
+          <h3>Sign In / Sign Up</h3>
+          <button
+            onClick={onClose}
+            className="modal-close-button"
+            title="Close"
+          >
+            ×
+          </button>
+        </div>
+
+        <Authenticator
+          loginMechanisms={['email']}
+          signUpAttributes={['name', 'email']}
+          formFields={{
+            signUp: {
+              name: {
+                order: 1,
+                placeholder: 'Enter your full name',
+                label: 'Full Name *',
+                required: true
+              },
+              email: {
+                order: 2,
+                placeholder: 'Enter your email address',
+                label: 'Email *'
+              },
+              password: {
```

Comment:
Are we able to show the password requirements here? It was a bit annoying that I had to keep submitting in order to see what was missing. Ie: uppercase, lowercase, special characters etc. 

<img width="507" height="743" alt="Image" src="https://github.com/user-attachments/assets/e252ce43-f2be-4c66-bf56-9c4698f248af" />